apps/back/src/Controller/ExpenseController.php

```diff
@@ -15,10 +16,26 @@
 #[Route('/expense', name: 'expense')]
 class ExpenseController extends AbstractController
 {
-    #[Route('/list', name: '_list')]
-    public function expensesList(): Response
+    #[Route('/list/{groupName}', name: '_list')]
+    public function expenseList(
+        EntityManagerInterface $em,
+        string $groupName,
+    ): Response
     {
-        return $this->render('expense/list.html.twig', []);
+        $groupRepository = $em->getRepository(Group::class);
+        $group = $groupRepository->findOneBy(['name' => $groupName]);
+
+        $expenseRepository = $em->getRepository(Expense::class);
+        $expenses = $expenseRepository->findBy(['group' => $group], ['date' => 'DESC']);
+
+        if($group !== null) {
+            return $this->render('expense/list.html.twig', [
+                'expenses' => $expenses,
+                'group' => $group,
+            ]);
+        } else {
+            return new Response(Response::HTTP_NOT_FOUND);
+        }
```

Comment:
Symfony provides a feature to achieve what you did here :)

You can do:

```suggestion
        if($group === null) {
            throw new HttpNotFoundException();
        }
        
        return $this->render('expense/list.html.twig', [
            'expenses' => $expenses,
            'group' => $group,
        ]);
```

Behind the hood, Symfony will catch this `HttpNotFoundException` and will automatically return a new `Response` with a 404 error code.

Reply:
Thanks, I didn't know. However with this change I get a `Undefined type 'App\Controller\HttpNotFoundException'.` warning